enzy_htp/_interface/pymol_interface.py

```diff
@@ -101,6 +102,111 @@ def select_pymol_obj(self, pattern: str, pymol_obj_name: str,
+                pymol_session.cmd.do("refresh_wizard")
+                pymol_session.cmd.get_wizard().set_mode(target)
+                if pymol_obj_name == "":
+                    pymol_session.cmd.get_wizard().do_select(f"{pos_key[0]}/{str(pos_key[1])}/")
+                else:
+                    pymol_session.cmd.get_wizard().do_select(f"{pymol_obj_name}//{pos_key[0]}/{str(pos_key[1])}/")
+            
+                pymol_session.cmd.get_wizard().do_state(i)
+                pymol_session.cmd.frame(i)
+                pymol_session.cmd.get_wizard().apply()
+                self.export_pymol_obj(pymol_obj_name, pymol_session, rotamer=i)
+            return
+
+        pymol_session.cmd.get_wizard().apply()
+
+    def export_pymol_obj(self, pymol_obj_name: str, 
+                        pymol_session: pymol2.PyMOL,
+                        if_retain_order: bool = True,
+                        rotamer: int = 0) -> str:
+        """
+        Saves a PyMOL object to a PDB file.
+        Args:
+            pymol_obj_name: the name of the target enzyme in PyMOL.
+            pymol_session: the target PyMOL session.
+            in_order: if the saving should keep the order of the atoms in the original object.
+            rotamer: the current rotamer of the mutation.
+        Returns:
+            The path to the saved PDB file.
+        """
+        result_dir = eh_config["system.SCRATCH_DIR"]
+        fs.safe_mkdir(result_dir)
+        if rotamer != 0:
+            pymol_outfile_path = fs.get_valid_temp_name(
+                    f"{result_dir}/{pymol_obj_name}_rotamer_{rotamer}.pdb")
+        else:
+            pymol_outfile_path = fs.get_valid_temp_name(
+                f"{result_dir}/{pymol_obj_name}.pdb")
+
+        if if_retain_order:
+            pymol_session.cmd.set("retain_order")
+        pymol_session.cmd.save(pymol_outfile_path, pymol_obj_name)
+
+        return pymol_outfile_path
+
+
+    def export_enzy_htp_stru(self, pymol_obj_name: str,
+                            pymol_session: pymol2.PyMOL,
+                            if_retain_order: bool = True) -> Structure:
+        """
+        Saves a PyMOL object to a Structure object.
+        Args:
+            pymol_obj_name: the name of the target enzyme in PyMOL.
+            pymol_session: the target PyMOL session.
+            in_order: if the saving should keep the order of the atoms in the original object.
+            keep_files: if the program should clean up the temp files or not.
+        Returns:
+            A Structure object representing the target enzyme in PyMOL.
+        """
+        sp = PDBParser()
+        temp_dir = eh_config["system.SCRATCH_DIR"]
+        fs.safe_mkdir(temp_dir)
+        pymol_outfile_path = fs.get_valid_temp_name(
+                f"{temp_dir}/pymol_output.pdb")
+
+        if if_retain_order:
+            pymol_session.cmd.set("retain_order")
+
+        pymol_session.cmd.save(pymol_outfile_path, pymol_obj_name)
```

Comment:
We can use the newly made export_pdb to avoid repeating the same code. (sorry if I commented too early that you were about to do this)